### ISSUE TITLE
external tab does n't back to APP when used with Firebase BackEnd ?

### ISSUE THREAD

**Issue by Joseph-Nathan** [ThexXTURBOXx/flutter_web_auth_2#5]
## Describe the bug
Firebase BackEnd does n't  support external callable method 

## To Reproduce
1. Create FireBase project 
2. provide any Auth Provider as Google
3. try to Authentication with external browser tab 
4. See Error :
`Unable to process request due to missing initial state. This may happen if browser sessionStorage is inaccessible or accidentally cleared.`

## Expected behavior
After Authentication , callable method fire and get back to App .

## Screenshots
![103522266-1f317180-4ea0-11eb-808f-50539f604b01](https://user-images.githubusercontent.com/60202661/187045222-a5d5296a-5ae5-4b7e-8ef4-f27252c8d816.jpeg)

## Device (please complete the following information!)
 - Device: [ OPPO F11 ]
 - OS: [Andriod 12]
 - Browser: [Chrome]
 - `flutter_web_auth` version: [e.g. 0.4.1]

## Additional context
if add support to in_App_Webview that will be great

**Comment by AndreiRoze** [comment]
Shortly, it's not library issue. look like it incorrect behaviour of Firebase for some phones/browsers. More info here https://github.com/firebase/firebase-js-sdk/issues/4256

**Comment by Joseph-Nathan** [comment]
@AndreiRoze we know , can you provide any way to pass this .

**Comment by AndreiRoze** [comment]
@Joseph-Nathan, you can try to use redirect to your server and from your server redirect to app by package name, but i am not sure, may be it useless.

**Comment by ThexXTURBOXx** [comment]
The issue may be related to Firebase, but the feature request with the In-App web view is a nice idea. So I let this open for now

**Comment by komaxx** [comment]
Wouldn't a In-App web view pretty much defeat the whole point of OAuth?

As far as I understand, running some sort of WebView in the caller's process also enables accessing everything that's entered into the login page (by adding JavaScript hooks, or DOM scraping, or piping interaction through an invisible overlay view, or ...), thus exposing user credentials to the calling app. 

At that point there would be no security gain over just letting the user enter her credentials directly into the app, right?

**Comment by ThexXTURBOXx** [comment]
Yes, that is the reason why I recently decided against it. However, I am still thinking about other ways to provide the callback process

**Comment by Joseph-Nathan** [comment]
any news for provide callback process to Firebase ?
